app/main/routes_tilt_api.py

```diff
@@ -0,0 +1,85 @@
+from webargs import fields
+from webargs.flaskparser import use_args, FlaskParser
+from marshmallow import INCLUDE, Schema
+import threading
+from datetime import datetime
+import json
+
+from . import main
+from .units import convert_temp
+from .. import socketio
+from .config import tilt_active_sessions_path
+from .model import TiltSession
+from .session_parser import active_tilt_sessions
+
+_lock = threading.Lock()
+arg_parser = FlaskParser()
+
+# This supports using pytilt to get the data from the tilt using bluetooth and send via this API call
+# See https://github.com/atlefren/pytilt but all you have to do is set the environment variable
+#   PYTILT_URL=<your picobrew_pico server>/API/tilt
+# and it will work without modifications
+
+class TiltSchema(Schema):
+    color = fields.Str(required=True),     # device ID
+    temp = fields.Float(required=True),    # device temperature in Celcius
+    gravity = fields.Int(required=True),   # calculated specific gravity
+    timestamp = fields.Str(required=True), # time sample taken in ISO format e.g '2021-03-06T16:25:42.823122'
+
+# Process tilt Data: /API/tilt
+@main.route('/API/tilt', methods=['POST'])
+@use_args(TiltSchema(many=True), unknown=INCLUDE)
+def process_tilt_dataset(dataset):
+    for data in dataset:
+        data['temp'] = convert_temp(data['temp'], 'F')
+        process_tilt_data(data)
+    return('', 200)
+
+# this can also be called from tilt.py if this is not running on a mac
+def process_tilt_data(data):
+    global _lock
+    with _lock:
+        uid = data['color']
+        
+        if uid not in active_tilt_sessions:
+            active_tilt_sessions[uid] = TiltSession()
+
+        if active_tilt_sessions[uid].active:
+            # initialize session and session files
+            if active_tilt_sessions[uid].uninit:
+                create_new_session(uid)
+
+            time = (datetime.fromisoformat(data['timestamp']) - datetime(1970, 1, 1)).total_seconds() * 1000
+            session_data = []
+            log_data = ''
+            point = {
+                'time': time,
+                'temp': data['temp'],
+                'gravity': (data['gravity'] / 1000) if data['gravity'] > 1000 else data['gravity'],
```

Comment:
why are you putting the `/ 1000` in a conditional looking for `> 1000`? I've had a number of dry beers that would have measurements broadcasted in the 900 range (0.9xx) as a final gravity.

Is there a way to change the gravity measurement of a tilt? If so then we should at least support `SG > 900 && SG < 1800` or something along those lines maybe just `> 900` is needed. Though if the measurement units aren't modifiable  (don't see how they would be as there isn't a UI) we should just always `/ 1000`, no?

Reply:
Good catch. it probably should always be divided by 1000. Not sure why I did that - it's not like something can go wrong.

Reply:
Some solutions I've seen have protections around large unexpected values. 

I've got some really strange values logged at times in google sheets:
- temps in the 150+ range (say when cleaning or even infrequent random invalid readings during fermentation)
- spikes of strange SG measurements (below 0.9 or higher than 1.8)

I bet PyTilt was limiting low end at 1.0 vs a more realistic 0.9x (for drier styles).

Reply:
![Screenshot_20210311-220248.png](https://user-images.githubusercontent.com/2158627/110885916-b4445280-82b5-11eb-824a-06df4766d626.png)

![Screenshot_20210311-220358.png](https://user-images.githubusercontent.com/2158627/110885920-b73f4300-82b5-11eb-978f-383e7213facb.png)

Reply:
no, it's just reporting the raw data from the tilt, which for me has always been over 1000. In plain water it returns 1010 right now.

Reply:
Ok, always dividing by 100 now, and rssi should be accurate